File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def dfs(graph, v, visited):
    visited[v] = True
    for i, g in enumerate(graph[v]):
        if g == 1 and i != v and visited[i] == False:
            dfs(graph, i, visited)

def solution(n, computers):
    answer = 0
    visited = [False] * len(computers)
    idx = 0
    
    while True:
        dfs(computers, idx, visited)
        answer += 1
        if visited.count(False) > 0:
            idx = visited.index(False)
        else:
            break

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
    boolean[] visited;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];
        for(int i=0;i<n;i++) {
            graph.add(new ArrayList<>());
            visited[i] = false;
        }
        for(int i=0;i<n;i++) {
            for(int j=i+1;j<n;j++) {
                if (computers[i][j] == 1) {
                    graph.get(i).add(j);
                    graph.get(j).add(i);
                }
            }
        }

        for(int i=0;i<n;i++) {
            if (visited[i] == false) {
                dfs(i);
                answer += 1;
            }
        }
        return answer;
    }
    
    public void dfs(int v) {
        visited[v] = true;
        for(int next : graph.get(v)) {
            if (visited[next] == false) {
                dfs(next);
            }
        }
    }
}